src/extension/globalExtensionState.ts

```diff
@@ -14,8 +14,9 @@ import { EventLogger, Severity } from "../logging/eventLogger";
 import { pluginId } from "./coqPilot";
 import VSCodeLogWriter from "./vscodeLogWriter";
 
-export class GlobalExtensionState {
+export class GlobalExtensionState implements Disposable {
     public readonly eventLogger: EventLogger = new EventLogger();
+    public hasActiveSession = true;
```

Comment:
Oh-oh, now it seems even more dangerous: a public modifiable property. A good reason to think about getting rid of it (by moving to another entity as `SessionState`, for example) or introducing a getter and a setter to make access more secure (by making it more explicit).